### PR TITLE
Gift Entry: Organize GE classes and remove FORM prefix

### PR DESCRIPTION
**Pull request by kenlewis-sfdo**
# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/classes/BGE_BatchGiftEntry_UTIL.cls [hunk 1]
```diff
@@ -103,13 +103,13 @@ public with sharing class BGE_BatchGiftEntry_UTIL {
             Map<String, BDI_FieldMapping> fieldMappingByDevName =
                     mappingService.fieldMappingByDevName;
 
-            FORM_Template t = (FORM_Template) JSON.deserialize(
+            GE_Template.Template t = (GE_Template.Template) JSON.deserialize(
```
Comment (voduyemi) [inline src/classes/BGE_BatchGiftEntry_UTIL.cls:106]: Can we rename this variable `t` to `template` or `formTemplate`

**Review by voduyemi — CHANGES_REQUESTED** [review]
Thanks @kenlewis-sfdo for making Gift Entry cleaner! I have a some comments around accessors and the `returnAuraException` method  in `GE_GiftEntryController` should be made `private`. But it looks so good.

File: src/classes/GE_Template_TEST.cls [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+            + 'permissions error should contain any field the user does not have access '
+            + 'to.');
+
+        */
+    }
+
+    private class UTIL_PermissionsMock implements StubProvider {
+
+        Boolean hasPermission;
+        Boolean hasCRUDPermission;
```
[the remark this answers is not in the document]
Comment (kenlewis-sfdo) [reply src/classes/GE_Template_TEST.cls:204]: 👍🏽

File: src/classes/GE_Template.cls [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+    }
+    
+    /**
+     * @description This method builds a mapping of partial dev name to 
+     * the full dev name using the org's field mapping custom metadata records. 
+     *
+     * @return A mapping of partial dev name to full dev name. 
+     * e.g. Account1_Name -> Account1_Name_ee63cf9df
+     */
+    public static void populateDevNameMapping(Map<String,String> updatedFieldDevNameMap,
```
[the remark this answers is not in the document]
Comment (kenlewis-sfdo) [reply src/classes/GE_Template.cls:212]: 👍🏽

File: src/classes/GE_Template.cls [hunk 4]
```diff
@@ -0,0 +1,590 @@
...
+    }
+
+    /**
+     * @description This method builds a new default template using the base JSON string. 
+     * It will update the base JSON string with the updated field mapping dev names.
+     *   
+     *
+     * @return A new Form template that is ready to be saved into the DB and used. 
+     */
+    public static Form_Template__c buildDefaultTemplate() {
```
[the remark this answers is not in the document]
Comment (kenlewis-sfdo) [reply src/classes/GE_Template.cls:112]: 👍🏽

File: src/classes/BGE_BatchGiftEntry_UTIL.cls (hunk 1 above)
Comment (kenlewis-sfdo) [reply src/classes/BGE_BatchGiftEntry_UTIL.cls:106]: 👍🏽

File: src/classes/GE_GiftEntryController.cls [hunk 5]
```diff
@@ -276,5 +279,774 @@ public with sharing class GE_GiftEntryController {
...
+        }
+        return template;
+    }
+
+    /*******************************************************************************************************
+    * @description Method for validating CRUD/FLS permissions for a template
+    * @param template - the template to use for CRUD/FLS validation
+    * @param accessLevels - a collection of CRUD/FLS validations to perform on the template (i.e. CREATE/UPDATE/READ)
+    */
+    public static GE_Template.Template checkPermissions(GE_Template.Template template, Set<GE_Template.AccessLevel> accessLevels){
```
[the remark this answers is not in the document]
Comment (kenlewis-sfdo) [reply src/classes/GE_GiftEntryController.cls:844]: 👍🏽

**Comment by kenlewis-sfdo** [comment]
Thanks for the review and comments @voduyemi!  I made those updates and merged this work into [the original PR](https://github.com/SalesforceFoundation/NPSP/pull/5846), which will be used to close this WI out.  Thanks again!

**Pull request closed without merging**